pkg/k8s/watchers/cilium_network_policy.go

```diff
@@ -159,6 +159,7 @@ func (k *K8sWatcher) ciliumNetworkPoliciesInit(ctx context.Context, cs client.Cl
 				case resource.Delete:
 					err = k.onDelete(slimCNP, cnpCache, event.Key, k8sAPIGroupCiliumClusterwideNetworkPolicyV2, resources.MetricCCNP)
 				}
+				reportCNPChangeMetrics(err)
```

Comment:
I guess this was a "bug" from the original code and we are fixing it now because the refactor exposed it, it's fine IMO. Typically, we'd want to separate this out into a separate PR, but I think it's pretty minor given that this was probably missing in the code for some time? Anyway, I don't think it's worth too much to look into this.

Reply:
Yes, it is. Being so small, I just added a separate commit in this PR to add the metric reporting also for CCNP.